### PR TITLE
Reduce the number of cache operations for dav operations

### PR DESCRIPTION
**Pull request by icewind1991**
- Reuse fileinfo from userfolder when possible
- cache root id in mountpoint
- cache userfolder in root
- optimize getUserFolder for the common case where the user is already setup

Reduces the number of cache operations done for the most basic (and common) dav operation (propfind with depth:0 on the root) from 7 to 2

### CODE REVIEW

File: tests/lib/TestCase.php [hunk 1]
```diff
@@ -124,6 +125,9 @@ protected function setUp() {
 				call_user_func([$this, $methodName]);
 			}
 		}
+
+
+		\OC::$server->getRootFolder()->clearCache();
```
Comment (nickvergessen) [inline tests/lib/TestCase.php]: This causes a `setupFS()` on each test case, even when the test doesn't even use anything from the file system. Shouldn't the call in `OC_Util::tearDownFS()` be enough?

Comment (icewind1991) [inline tests/lib/TestCase.php]: Good catch, fixed

**Comment by rullzer** [comment]
Nice catch. I doesn't read very easy currently. but that is mainy because of the rest of the code around it.

:+1:

**Comment by LukasReschke** [comment]
LGTM

**Pull request merged**